packages/url-launcher/ios/Classes/UrlLauncherPlugin.m

```diff
@@ -3,28 +3,56 @@
 @implementation UrlLauncherPlugin {
 }
 
-- (instancetype)initWithController:(FlutterViewController *)controller {
+- (instancetype)initWithController:(FlutterViewController*)controller {
   self = [super init];
   if (self) {
     FlutterMethodChannel* channel = [FlutterMethodChannel
                                      methodChannelWithName:@"plugins.flutter.io/URLLauncher"
                                      binaryMessenger:controller];
     [channel setMethodCallHandler:^(FlutterMethodCall *call,
                                     FlutterResult result) {
-      if ([@"UrlLauncher.launch" isEqualToString:call.method]) {
-        [self launchURL:call.arguments];
-        result(nil);
+      NSString* url = call.arguments;
+      if ([@"UrlLauncher.canLaunch" isEqualToString:call.method]) {
+        result(@([self canLaunchURL:url]));
+      } else if ([@"UrlLauncher.launch" isEqualToString:call.method]) {
+        [self launchURL:url result:result];
+      } else {
+        result(FlutterMethodNotImplemented);
       }
     }];
   }
   return self;
 }
 
-- (NSDictionary*)launchURL:(NSString*)urlString {
+- (BOOL)canLaunchURL:(NSString*)urlString {
   NSURL* url = [NSURL URLWithString:urlString];
   UIApplication* application = [UIApplication sharedApplication];
-  bool success = [application canOpenURL:url] && [application openURL:url];
-  return @{@"success": @(success) };
+  return [application canOpenURL:url];
+}
+
+- (void)launchURL:(NSString*)urlString result:(FlutterResult)result {
+  NSURL* url = [NSURL URLWithString:urlString];
```

Comment:
According to https://developer.apple.com/reference/foundation/nsurl/1572047-urlwithstring url is nil.